lib/Backend/arm64/EncoderMD.cpp

```diff
@@ -888,10 +888,12 @@ EncoderMD::GenerateEncoding(IR::Instr* instr, BYTE *pc)
         break;
 
     case Js::OpCode::LDR:
+        Assert(instr->GetSrc1()->GetSize() >= 4 || instr->GetSrc1()->IsUnsigned());
         bytes = this->EmitLoadStore(Emitter, instr, instr->GetSrc1(), instr->GetDst(), EmitLdrbRegister, EmitLdrhRegister, EmitLdrRegister, EmitLdrRegister64, EmitLdrbOffset, EmitLdrhOffset, EmitLdrOffset, EmitLdrOffset64);
```

Comment:
Why the GetSize() check? LDR can also be used to load bytes and half-words.

Reply:
I also notice LDRS doesn't have the size check.

Reply:
The assert is that IF a byte or half-word is used, it needs to be unsigned because the destination is guaranteed to be 4+ bytes. I added the size check because TyMachReg is signed and we use it all indiscriminately in the machine independent code. I suppose it would be more appropriate to check ``instr()->GetDst()->GetSize() <= instr()->Src1()->GetSize() || instr->GetSrc1()->IsUnsigned()`` rather than limiting the check to handle byte and half-word cases.

Reply:
Ok, I see it now. Guess I misread. And technically you're right about checking the size against the dest size.